app/scripts/metamask-controller.js

```diff
@@ -3367,38 +3365,12 @@ export default class MetamaskController extends EventEmitter {
       await new Promise((resolve) => setTimeout(resolve, 5_000));
     }
 
-    const [primaryKeyring] = this.keyringController.getKeyringsByType(
-      KeyringType.hdKeyTree,
-    );
-    if (!primaryKeyring) {
-      throw new Error('MetamaskController - No HD Key Tree found');
-    }
-    const { keyringController } = this;
-    const { identities: oldIdentities } =
-      this.preferencesController.store.getState();
-
-    if (Object.keys(oldIdentities).length === accountCount) {
-      const oldAccounts = await keyringController.getAccounts();
-      const keyState = await keyringController.addNewAccount(primaryKeyring);
-      const newAccounts = await keyringController.getAccounts();
-
-      await this.verifySeedPhrase();
-
-      this.preferencesController.setAddresses(newAccounts);
-      newAccounts.forEach((address) => {
-        if (!oldAccounts.includes(address)) {
-          this.preferencesController.setSelectedAddress(address);
-        }
-      });
+    const { addedAccountAddress } =
+      await this.coreKeyringController.addNewAccount(accountCount);
 
-      const { identities } = this.preferencesController.store.getState();
-      return { ...keyState, identities };
-    }
+    this.preferencesController.setSelectedAddress(addedAccountAddress);
```

Comment:
I'm noticing that the old implementation only changed the selected address if the added account was _new_. Perhaps this behavior should be preserved for now?

Reply:
Uhm, it seems like the CI is failing now, even if I think I used the same check as before. I'm not sure what's the problem here

Reply:
I fixed it using the response from `coreKeyringController.getAccounts()` to keep track of old accounts instead of the identities from the preferences controller